### PR TITLE
turn useONNXModelType default to true

### PR DESCRIPTION
**Pull request by chentong319**
Rely on shape inference of onnx, at least for type inference.
Clean up the code of our type map for onnx op.

### CODE REVIEW

**Comment by chentong319** [comment]
It turned out that I can not clean up the typeMap for onnx Ops yet. The CustomOp needs it. More change in the FrontendDialectTransform.cpp is needed.

File: src/Builder/FrontendDialectTransformer.cpp [hunk 1]
```diff
@@ -1386,6 +1388,17 @@ bool ImportFrontendModelInternal(onnx::ModelProto &model, MLIRContext &context,
...
+  // Note: when options.useOnnxModelTypes is true, the onnx::shape_inference
+  // cannot handle non onnx operations (represented as CustomOp in onnx-mlir)
+  // Assertion error if the model contains a such operation.
+  // onnx-mlir handleds the CustomOp in a different way. It assumes the common
+  // pattern that the element type of the output is the same
+  // as the the first input. And later shape inference will use the
+  // shape-inference-pattern attribute to perform shape inference on CustomOp.
+  // The type assumption of the Importer may be incorrect and cause
+  // trouble.
+  // ToFix: the shape-nference-pattern should be added and used in Importer.
```
Comment (sorenlassen) [inline src/Builder/FrontendDialectTransformer.cpp:1400]: typo: should be "shape-inference-pattern"

File: src/Builder/FrontendDialectTransformer.cpp [hunk 2]
```diff
@@ -1386,6 +1388,17 @@ bool ImportFrontendModelInternal(onnx::ModelProto &model, MLIRContext &context,
     }
   }
 
+  // Note: when options.useOnnxModelTypes is true, the onnx::shape_inference
+  // cannot handle non onnx operations (represented as CustomOp in onnx-mlir)
+  // Assertion error if the model contains a such operation.
+  // onnx-mlir handleds the CustomOp in a different way. It assumes the common
```
Comment (sorenlassen) [inline src/Builder/FrontendDialectTransformer.cpp:1394]: typo: "handles"

**Review by sorenlassen — APPROVED** [review]
LGTM

after this lands, please check if there are any model zoo regressions (or you could potentially check beforehand if you know how)

**Pull request merged**